src/engine-actions/validate-node.js

```diff
@@ -0,0 +1,78 @@
+const {
+  getInfo,
+  genSeed
+} = require('../lnd-actions')
+
+const validateNodeConfig = require('./validate-node-config')
+
+/**
+ * CODE 12 for gprc is equal to 'unimplemented'
+ * @see https://github.com/grpc/grpc-go/blob/master/codes/codes.go
+ * @constant
+ * @type {Number}
+ * @default
+ */
+const UNIMPLEMENTED_SERVICE_CODE = 12
+
+/**
+ * Rough estimate if the lnd instance's wallet is unlocked or not
+ *
+ * @private
+ * @param {grpc.Client} Lightning grpc client
+ * @return {Boolean} isEngineUnlocked
+ */
+async function isEngineUnlocked (client) {
+  try {
+    // If the call to `getInfo` succeeds, then we can assume that our LND instance
+    // is unlocked and functional
+    await getInfo({ client })
+    return true
+  } catch (e) {
+    // CODE 12 for grpc is equal to 'unknown service' or an error type of
+    // unimplemented.
+    // In GRPC Unimplemented indicates operation is not implemented or not
+    // supported/enabled in this specific service.
+    if (e.code && e.code === UNIMPLEMENTED_SERVICE_CODE) {
+      return false
+    }
+
+    // We return true, because the error is unrelated to a service not being implemented
+    // and the user will now have to troubleshoot further
+    return true
```

Comment:
It seems weird to return true if we don't know whether the engine is unlocked or not. Maybe it's just semantics, but it feels like this is more of an unknown state.

Reply:
@treygriffith I was worried about generic errors from LND. An example is LND throwing an error `block out of order`... `getInfo` would fail but the engine is technically unlocked.

Reply:
@treygriffith in the recent commits, we always return `FALSE` unless we can prove that the `Lightning` RPC is up.